clarity/src/vm/functions/arithmetic.rs

```diff
@@ -441,6 +482,64 @@ pub fn native_mod(a: Value, b: Value) -> InterpreterResult<Value> {
     type_force_binary_arithmetic!(modulo, a, b)
 }
 
+pub fn native_bitwise_left_shift(input: Value, pos: Value) -> InterpreterResult<Value> {
+    if let Value::UInt(u128_val) = pos {
+        let shamt = u32::try_from(u128_val & 0xffffffff)
+            .expect("FATAL: lower 32 bits did not convert to u32");
+
+        match input {
+            Value::Int(input) => {
+                let result = input
+                    .wrapping_shl(shamt);
+                Ok(Value::Int(result))
+            },
+            Value::UInt(input) => {
+                let result = input
+                    .wrapping_shl(shamt);
+                Ok(Value::UInt(result))
+            },
+            _ => Err(CheckErrors::UnionTypeError(
+                vec![ 
+                    TypeSignature::IntType, 
+                    TypeSignature::UIntType 
+                ], 
+                TypeSignature::type_of(&input))
+                .into())
+        }
+    } else {
+        Err(CheckErrors::TypeValueError(TypeSignature::UIntType, pos).into())
+    }
+}
+
+pub fn native_bitwise_right_shift(input: Value, pos: Value) -> InterpreterResult<Value> {
+    if let Value::UInt(u128_val) = pos {
+        let shamt = u32::try_from(u128_val & 0xffffffff)
```

Comment:
See comment as above about masking with `0x7f` to be more explicit about the behavior.